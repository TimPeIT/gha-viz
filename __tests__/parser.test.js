const path = require("path");
const { loadYaml, parseWorkflow, renderText, renderMermaid } = require("../index");

test("lÃ¤dt YAML und parst Jobs/Needs/Steps korrekt", () => {
  const doc = loadYaml(path.join(__dirname, "..", "samples", "ci.yml"));
  const model = parseWorkflow(doc);
  const jobNames = model.items.map(j => j.name);
  expect(jobNames).toEqual(["build","test","lint"]);
  const testJob = model.items.find(j => j.name === "test");
  expect(testJob.needs).toEqual(["build"]);
  expect(testJob.stepSummaries[0]).toMatch(/run\s+npm test/);
});

test("Snapshot der Textausgabe", () => {
  const doc = loadYaml(path.join(__dirname, "..", "samples", "ci.yml"));
  const model = parseWorkflow(doc);
  const text = renderText(model);
  expect(text).toMatchSnapshot();
});

test("Mermaid enthÃ¤lt Kanten", () => {
  const doc = loadYaml(path.join(__dirname, "..", "samples", "ci.yml"));
  const model = parseWorkflow(doc);
  const mmd = renderMermaid(model);
  expect(mmd).toMatch(/build\s*-->\s*test/);
});
