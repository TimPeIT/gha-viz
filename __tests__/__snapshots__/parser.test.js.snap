// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot der Textausgabe 1`] = `
"== Triggers ==
- push
- pull_request

== Jobs ==
* build
  needs: (none)
  steps:
    - #1 uses actions/checkout@v4
    - #2 run  npm ci
* test
  needs: build
  steps:
    - #1 run  npm test
* lint
  needs: build
  steps:
    - #1 run  npm run lint

== Dependencies (ASCII) ==
- build -> test
- build -> lint"
`;
